meta-mel/conf/distro/include/mel.conf

```diff
@@ -73,7 +73,7 @@ META_MENTOR_DIR = "${LAYERDIR_mel}/.."
 PATH_prepend = "${META_MENTOR_DIR}/scripts:"
 
 # Default to the external toolchain if available
-TCMODE ?= "${@bb.utils.contains('BBFILE_COLLECTIONS', 'sourcery', 'external-sourcery', bb.utils.contains('BBFILE_COLLECTIONS', 'external-toolchain', 'external', 'default', d), d)}"
+TCMODE ?= "${@bb.utils.contains('BBFILE_COLLECTIONS', 'sourcery', 'external-sourcery-rebuild-libc', bb.utils.contains('BBFILE_COLLECTIONS', 'sourcery', 'external-sourcery-rebuild-libc', 'default', d), d)}"
```

Comment:
in both conditions you are looking for sourcery layer and returning same value. Kindly see the line which you have modified.